src/test/java/com/hotels/bdp/circustrain/bigquery/extraction/BigQueryDataExtractionManagerTest.java

```diff
@@ -88,7 +133,46 @@ public void locationTest() {
   }
 
   @Test
-  public void locationForTableThatHasntBeenExtractedIsNull() {
-    assertNull(dataExtractionManager.location(table));
+  public void locationForTableThatHasntBeenExtractedCachesAndReturnsLocation() {
+    assertNotNull(dataExtractionManager.location(table));
+  }
+
+  @Test
+  public void tableWrapperHashCodeTest() {
+    BigQueryDataExtractionManager.TableWrapper wrapperOne = new BigQueryDataExtractionManager.TableWrapper(table);
+    BigQueryDataExtractionManager.TableWrapper wrapperTwo = new BigQueryDataExtractionManager.TableWrapper(table);
+
+    assertEquals(wrapperOne.hashCode(), wrapperTwo.hashCode());
+  }
+
+  @Test
+  public void tableWrapperEqualsTest() {
+    BigQueryDataExtractionManager.TableWrapper wrapperOne = new BigQueryDataExtractionManager.TableWrapper(table);
+    BigQueryDataExtractionManager.TableWrapper wrapperTwo = new BigQueryDataExtractionManager.TableWrapper(table);
+    assertEquals(wrapperOne, wrapperTwo);
+  }
+
+  @Test
+  public void tableWrapperNotEqualsTest() {
+    BigQueryDataExtractionManager.TableWrapper wrapperOne = new BigQueryDataExtractionManager.TableWrapper(table);
+    Table mock = mock(Table.class);
+    when(mock.getTableId()).thenReturn(TableId.of("dataset2", "table2"));
+    BigQueryDataExtractionManager.TableWrapper wrapperTwo = new BigQueryDataExtractionManager.TableWrapper(mock);
+    assertNotEquals(wrapperOne, wrapperTwo);
+  }
+
+  @Test
+  public void tableWrapperInSetTest() {
```

Comment:
If you've already tested the equals methods works as expected then you don't need to also test how it works in a Set,  we can assume how the Set uses equals() has already been tested inside the JDK.